server/controllers/campaigns.js

```diff
@@ -70,3 +70,57 @@ export function getAllCampaigns(req, res) {
     });
 }
 
+
+function validCampaignStatusUpdate(prevStatus, currStatus) {
+  const validCampaignStatusTransitions = {
+    draft: 'active',
+    active: ['pause', 'completed'],
+    pause: ['active', 'completed'],
+    completed: 1
+  };
+  const validStatusOptions = Object.keys(validCampaignStatusTransitions);
+  let validTransitionForStatus;
+  if (prevStatus === 'completed') {
+    throw new Error('Campaigns with status set to "completed" may not be changed.');
+  }
+  if (!validCampaignStatusTransitions[currStatus]) {
+    throw new Error(`Status update to: ${currStatus} is invalid, valid status options: ${validStatusOptions.join(', ')}.`);
+  }
+  if (prevStatus === 'active' || prevStatus === 'pause') {
+    validTransitionForStatus = validCampaignStatusTransitions[prevStatus].join(', ');
+    if (!validCampaignStatusTransitions[prevStatus].includes(currStatus)) {
+      throw new Error(`Invalid campaign status transition. Status ${prevStatus} can only transition to ${validTransitionForStatus} `);
+    }
+  } else if (validCampaignStatusTransitions[prevStatus] !== currStatus) {
+    validTransitionForStatus = validCampaignStatusTransitions[prevStatus];
+    throw new Error(`Invalid campaign status transition. Status ${prevStatus} can only transition to ${validTransitionForStatus} `);
+  }
+  return true;
+}
+
+export function updateCampaignById(req, res) {
+  const campaign_id = parseInt(req.params.id, 10);
+  const { status: newStatus, name, title, description, contact_lists_id, script_id } = req.body;
+  return campaignsService.getCampaignById({ id: campaign_id })
+    .then((campaignObj) => {
+      const { status: previousStatus, id } = campaignObj.attributes;
+      try {
+        validCampaignStatusUpdate(previousStatus, newStatus);
+      } catch (error) {
+        return res.status(400).json({ message: `Invalid status transition, campaign status not updated: ${error}` });
+      }
+      return campaignsService.updateCampaignById(
```

Comment:
I feel we should add an additional check to say that name, title, description, contact list and script ID can only be changed when the campaign is a draft. Meaning that once we start the campaign (regardless of pause, active, inactive status), we should not be able to mutate any questions/contacts etc. What do you think?

In other words, how about we only allow status mutations for non-draft campaigns?